src/app/shell/personal-cabinet/provider/create-workshop/create-about-form/working-hours-form-wrapper/working-hours-form/working-hours-form.component.ts

```diff
@@ -129,8 +150,97 @@ export class WorkingHoursFormComponent implements OnInit, OnDestroy {
     });
   }
 
+  public onStartTimeSet(chosenTime: string): void {
+    this.startTimeFormControl.setValue(chosenTime);
+  }
+
+  public onEndTimeSet(chosenTime: string): void {
+    this.endTimeFormControl.setValue(chosenTime);
+  }
+
   public ngOnDestroy(): void {
     this.destroy$.next(true);
     this.destroy$.unsubscribe();
   }
+
+  public onStartBlur(): void {
+    if (!this.startTimeFormControl.value) {
+      this.startTimeFormControl.setValue(this.getMaxTime());
+    }
+  }
+
+  public onEndBlur(): void {
+    if (!this.endTimeFormControl.value) {
+      this.endTimeFormControl.setValue(this.getMinTime());
+    }
+  }
+
+  public validateTimeInput(event: Event): void {
+    const input = event.target as HTMLInputElement;
+    const cleanedValue = input.value.replace(TIME_REGEX_REPLACE, '');
+    input.value = cleanedValue;
+  }
+}
+
+export function timeFormatValidator(): ValidatorFn {
+  return (control: AbstractControl): ValidationErrors | null => {
+    const value = control.value;
+    if (!value) {
+      return null;
+    }
+
+    const valid = TIME_FORMAT_REGEX.test(value);
+    return valid ? null : { invalidTimeFormat: true };
+  };
+}
+
+export function timeRangeValidator(startCtrlName: string = 'startTime', endCtrlName: string = 'endTime'): ValidatorFn {
+  return (formGroup: AbstractControl): ValidationErrors | null => {
+    const startCtrl = formGroup.get(startCtrlName);
+    const endCtrl = formGroup.get(endCtrlName);
+
+    if (!startCtrl || !endCtrl) {
+      return null;
+    }
+
+    const startTime = startCtrl.value;
+    const endTime = endCtrl.value;
+
+    removeControlErrorKey(startCtrl, 'invalidTimeRange');
```

Comment:
It's propably should be enough to return `{ invalidTimeRange: true }` in case of error or `null`